### PR TITLE
fix(clv2): align Python _update_registry schema with shell counterpart

### LINKED ISSUE #2299: bug: Python _update_registry omits 'id' field present in shell counterpart
## Summary

In `skills/continuous-learning-v2/scripts/instinct-cli.py` around line 419, the Python `_update_registry` function writes registry entries without an `"id"` field:

```python
registry[pid] = {
    "name": pname,
    "root": proot,
    "remote": premote,
    "last_seen": datetime.now(timezone.utc).isoformat().replace("+00:00", "Z"),
}
```

However, the shell counterpart in `scripts/detect-project.sh` (around line 292) includes the `"id"` field:

```python
metadata = {
    "id": os.environ["_CLV2_REG_PID"],
    "name": os.environ["_CLV2_REG_PNAME"],
    "root": os.environ["_CLV2_REG_PROOT"],
    "remote": os.environ["_CLV2_REG_PREMOTE"],
    "created_at": entry.get("created_at", now),
    "last_seen": now,
}
```

## Impact

- Registry entries created via the Python path lack the `"id"` field
- Registry entries created via the shell path also include `"created_at"` which the Python path omits
- Code that reads the registry and expects `"id"` or `"created_at"` to be present will behave differently depending on which path wrote the entry
- The registry key is the project ID, so the `"id"` field is technically redundant, but the inconsistency can cause confusion

## Suggested fix

Align the Python `_update_registry` to include the same fields as the shell version, or vice versa. The shell version is more complete (has `id` and `created_at`), so updating the Python version to match is the safer direction.

### PR DESCRIPTION
**Pull request by gaurav0107**
## What Changed
Aligned the Python `_update_registry` writer in
`skills/continuous-learning-v2/scripts/instinct-cli.py` with its shell
counterpart in `detect-project.sh`. The registry entry now carries the same
field set and order: `id`, `name`, `root`, `remote`, `created_at`,
`last_seen`. `created_at` is stamped on first write and preserved on
subsequent updates; only `last_seen` advances.

## Why This Change
Previously the Python path wrote only `name`, `root`, `remote`, `last_seen`,
while the shell path wrote `id` and `created_at` as well. A `projects.json`
entry therefore had a different shape depending on which path (Python CLI or
shell hook) last touched it. Readers expecting `created_at` (or `id`) got
inconsistent results. This makes both writers produce identical schemas.

## Testing Done
- [x] `pytest test_parse_instinct.py` — 70 passed (added 2 focused tests:
  field-parity and `created_at` preservation across updates)
- [x] `node scripts/ci/validate-skills.js` — 271 skill dirs validated
- [x] `node scripts/ci/validate-no-personal-paths.js` — clean
- [x] `node scripts/ci/check-unicode-safety.js` — clean
- Edge cases considered: first write stamps `created_at`; re-write preserves it while refreshing `last_seen`; missing/corrupt registry falls back to `{}` (unchanged behavior)

## Type of Change
- [x] `fix:` Bug fix

## Security & Quality Checklist
- [x] No secrets or API keys committed
- [x] JSON files validate cleanly
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## Documentation
- [x] Added an inline comment explaining the shell-parity intent
- [x] No README change needed (internal registry schema)

Fixes #2299

### CODE REVIEW

File: skills/continuous-learning-v2/scripts/instinct-cli.py [hunk 1]
```diff
@@ -431,11 +431,20 @@ def _update_registry(pid: str, pname: str, proot: str, premote: str) -> None:
...
+        # existing entry; only "last_seen" advances on update.
+        now = datetime.now(timezone.utc).isoformat().replace("+00:00", "Z")
+        existing = registry.get(pid, {})
         registry[pid] = {
+            "id": pid,
             "name": pname,
             "root": proot,
             "remote": premote,
-            "last_seen": datetime.now(timezone.utc).isoformat().replace("+00:00", "Z"),
+            "created_at": existing.get("created_at", now),
```
[the remark this answers is not in the document]
Comment (gaurav0107) [reply skills/continuous-learning-v2/scripts/instinct-cli.py:457]: Good catch. Addressed in aa9af051: existing is now normalized to {} when it is not a dict, so a malformed (non-dict) registry value for the project id is healed by the rewrite instead of crashing the update. Added test_update_registry_heals_malformed_entry to cover it. Thanks for the review.

[the remark this answers is not in the document]
Comment (gaurav0107) [reply skills/continuous-learning-v2/scripts/instinct-cli.py:457]: Already addressed in aa9af051 — existing is coerced to {} when it is not a dict, so a malformed projects.json entry is healed rather than crashing the update.

File: skills/continuous-learning-v2/scripts/test_parse_instinct.py [hunk 2]
```diff
@@ -1047,6 +1047,38 @@ def test_update_registry_atomic_replaces_file(patch_globals):
...
+    tree = patch_globals
+    _update_registry("abc123", "demo", "/repo", "https://example.com/repo.git")
+    first = json.loads(tree["registry_file"].read_text())["abc123"]
+
+    _update_registry("abc123", "demo-renamed", "/repo", "https://example.com/repo.git")
+    second = json.loads(tree["registry_file"].read_text())["abc123"]
+
+    assert second["created_at"] == first["created_at"]
+    assert second["name"] == "demo-renamed"
+    assert second["last_seen"] >= first["last_seen"]
```
[the remark this answers is not in the document]
Comment (gaurav0107) [reply skills/continuous-learning-v2/scripts/test_parse_instinct.py:1082]: Addressed in 826ab424. Added an explicit assert that created_at == last_seen on the initial write (both derive from the same now), and split the compound assertions. Note: this repo's ruff config selects E/F/I/N/W/UP only, so PT018 is not enabled here, but the explicit contract assertion is a worthwhile improvement regardless.

**Review by daltino — APPROVED** [review]
The PR ensures alignment between the Python `_update_registry` method and its shell script counterpart by handling edge cases (e.g., non-dict registries) and adding schema validations through additional test coverage. The implementation appears clean and consistent with contribution guidelines, and the added test ensures the schema parity. No major issues spotted—nice work!

File: skills/continuous-learning-v2/scripts/test_parse_instinct.py [hunk 3]
```diff
@@ -1047,6 +1047,70 @@ def test_update_registry_atomic_replaces_file(patch_globals):
...
+    # crash _update_registry before the per-entry guard runs. The whole file is
+    # healed by the rewrite, preserving the old unconditional-overwrite behavior.
+    tree = patch_globals
+    tree["registry_file"].write_text(json.dumps(["oops"]), encoding="utf-8")
+    _update_registry("abc123", "demo", "/repo", "https://example.com/repo.git")
+    registry = json.loads(tree["registry_file"].read_text())
+    assert isinstance(registry, dict)
+    entry = registry["abc123"]
+    assert entry["id"] == "abc123"
+    assert entry["created_at"] == entry["last_seen"]
```
[the remark this answers is not in the document]
Comment (gaurav0107) [reply skills/continuous-learning-v2/scripts/test_parse_instinct.py:1111]: Declining this nit, after verifying it against the repo config.

1. The `unit` marker isn't registered for this test module. It's only registered in `tests/conftest.py` via `config.addinivalue_line("markers", "unit: ...")`, and that conftest doesn't cover `skills/continuous-learning-v2/scripts/` — there is no conftest on the path to this file. I confirmed by running pytest: adding `@pytest.mark.unit` here emits `PytestUnknownMarkWarning: Unknown pytest.mark.unit`. The repo's `filterwarnings` only ignores `DeprecationWarning`, so it would not be suppressed.

2. None of the 72 tests in this file use `@pytest.mark`, a `-> None` return annotation, or typed fixture params. Adding them to just this one test would be inconsistent with the module's established style, so I'm leaving the signature as-is to match its siblings.

Happy to revisit if `unit` is later registered for this path (e.g. a conftest under `skills/`), but that's out of scope for this fix.
